xds/internal/xdsclient/xdsresource/unmarshal_cds.go

```diff
@@ -70,9 +78,12 @@ const (
 
 func validateClusterAndConstructClusterUpdate(cluster *v3clusterpb.Cluster) (ClusterUpdate, error) {
 	var lbPolicy *ClusterLBPolicyRingHash
+	var lbCfgJSON json.RawMessage
```

Comment:
Your reading is correct. However, this config and subsequent documentation are about the ring hash layer. The xDS layer assumes 1024 default min and 1024 * 1024 * 8 default max (see this line in A42: "Note that gRPC has
different restrictions on ring size than Envoy does and a different default
for the max ring size (see below for details), so to ensure correct xDS
defaults, the xDS code must explicitly set this field in the gRPC LB policy
config even if the field is unset in the xDS resource (and therefore the
default xDS value of 8388608 is used)"). Thus, at the xDS layer, it will emit 8 million from the client, and our default ring hash cap will bring it down to 4096 (unless set by env).

Reply:
This also applies to your comment in converter.go about the min and max ring size (for some reason Github isn't allowing me to reply to that one).

Reply:
Confirmed with Java by the way, this is what they're doing and this is the correct behavior at the xDS layer.